src/main/java/com/mraof/minestuck/world/gen/feature/SulfurPoolFeature.java

```diff
@@ -0,0 +1,43 @@
+package com.mraof.minestuck.world.gen.feature;
+
+import com.mojang.serialization.Codec;
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.WorldGenLevel;
+import net.minecraft.world.level.levelgen.Heightmap;
+import net.minecraft.world.level.levelgen.feature.Feature;
+import net.minecraft.world.level.levelgen.feature.FeaturePlaceContext;
+import net.minecraft.world.level.levelgen.feature.configurations.NoneFeatureConfiguration;
+import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate;
+
+public class SulfurPoolFeature extends Feature<NoneFeatureConfiguration>
+{
+	private static final ResourceLocation SULFUR_POOL = new ResourceLocation(Minestuck.MOD_ID, "sulfur_pool");
+	private static final ResourceLocation LARGE_SULFUR_POOL = new ResourceLocation(Minestuck.MOD_ID, "large_sulfur_pool");
+	
+	public SulfurPoolFeature(Codec<NoneFeatureConfiguration> codec)
+	{
+		super(codec);
+	}
+	
+	private static ResourceLocation pickTemplate(RandomSource random)
+	{
+		return random.nextBoolean() ? SULFUR_POOL: LARGE_SULFUR_POOL;
+	}
+	@Override
+	public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context)
+	{
+		WorldGenLevel level = context.level();
+		RandomSource rand = context.random();
+		StructureTemplate template = context.level().getLevel().getStructureManager().getOrCreate(pickTemplate(rand));
+		BlockPos centerPos = context.origin();
+		TemplatePlacement placement = TemplatePlacement.centeredWithRandomRotation(template, centerPos, rand);
+		
+		int y = placement.minHeight(Heightmap.Types.WORLD_SURFACE_WG, level);
+		placement.placeAt(y, context);
+		
+		return true;
+	}
```

Comment:
This could benefit from the type of early return statements used in FrogRuinFeature and TreeStumpFeature. It looks a little peculiar when it hangs off the edge of a hill, the early return would require it to meet a certain flatness threshold in order to actually place it down. 

It would also stop it from generating inside the ground

```
if(placement.heightRange(Heightmap.Types.OCEAN_FLOOR, context.level()).difference() > 3)
			return false;
```

Reply:
agree, what line would i put the code you provided in this case?

Reply:
Between the creation of the "placement" variable and the creation of the "y" variable

Reply:
cool. will this still allow the pools to generate on top of/next to each other? that's something that i like a lot since it makes it a bit more variable when it happens

Reply:
It may happen slightly less often since a larger area would have to be flat before anything is placed down, but it should still be fine